packages/babel-plugin/src/utils/ast-builders.tsx

```diff
@@ -145,6 +145,109 @@ const traverseStyledBinaryExpression = (node: t.BinaryExpression, nestedVisitor:
   return node;
 };
 
+/**
+ * Handles cases like:
+ * 1. `propz.loading` in `border-color: \${(propz) => (propz.loading ? colors.N100 : colors.N200)};`
```

Comment:
nit: could we handle this in an `enter()` visitor instead? so the logic of "rename propz to props" only needs to function once - instead of needing to traverse up the parent

🤔 thinking about it i see we check for arrow function or binary expression on L273/L277. perhaps this is fine?

Reply:
So we are already traversing down. Here we are making sure `propz` is coming from function params. We picked member expression property, traversed to make sure it is coming from function params, and renamed it. Earlier we hardcoded the check that value should be `props`.